src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -116,24 +129,65 @@ class CollectorService(
             null
         }
 
-        eventIdToLastCheckTask.compute(CheckTaskKey(chainID, request.connectivityId)) { _, value ->
+        rules.compute(ruleId) { _, value ->
```

Comment:
Why have you changed that? We need to be able to create a chain with rules. By now we use only rule ID that is unique for every rule

Reply:
Because CheckTaskKey keys have never been used. And this map was used as simple collection (values were accessed only through an iterator).

Reply:
Because CheckTaskKey keys have never been used. And this map was used as simple collection (values were accessed only through an iterator).

Reply:
It is used to compute the same key when a new rule with the same chain ID and connectivity ID is submitted (the new rule will be added to the existing rule to start right after the previous one is finished). Please, return the map back.